pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,343 @@
+		listBindings: context.RBAC.ClusterRoleBindings("").Controller().Lister(),
+		bindings:     context.RBAC.ClusterRoleBindings(""),
+		policies:     context.Extensions.PodSecurityPolicies(""),
+		listRoles:    context.RBAC.ClusterRoles("").Controller().Lister(),
+		roles:        context.RBAC.ClusterRoles(""),
+		namespaces:   context.Core.Namespaces("").Controller().Lister(),
+		projects:     context.Management.Management.Projects("").Controller().Lister(),
+	}
+
+	context.Core.ServiceAccounts("").AddHandler(
+		"universalServiceAccountHandler",
+		func(key string, obj *v1.ServiceAccount) error {
+			logrus.Info("handle service account")
+
+			if obj == nil {
+				logrus.Infof("no service account provided, exiting")
+				// do nothing
+				return nil
+			}
+
+			namespaceName := obj.Namespace
+
+			logrus.Infof("svc acct name: %v namespace: %v", obj.Name, namespaceName)
+
+			// Check if a PSP exists for this cluster
+			policies, err := m.listPolicies.List("", labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting policies: %v", err)
+			}
+
+			bindings, err := m.listBindings.List(namespaceName, labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting cluster role bindings: %v", err)
+			}
+
+			var policy *v1beta12.PodSecurityPolicy
+
+			for _, candidate := range policies {
+				if candidate.ObjectMeta.Annotations[podSecurityAnnotation] == "true" {
+					policy = candidate
+				}
+			}
+
+			if policy == nil {
+				// get PSPT
+				// if no PSPT then get default
+				// if no default then exit
+				namespaces, err := m.namespaces.List("", labels.Everything())
+
+				if err != nil {
+					return fmt.Errorf("error getting namespaces: %v", err)
+				}
+
+				var namespace *v1.Namespace
+
+				for _, candidate := range namespaces {
+					if candidate.Name == namespaceName {
+						namespace = candidate
+					}
+				}
+
+				if namespace == nil {
+					return fmt.Errorf("no namespace exists with namespace of service account: %v", namespaceName)
+				}
+
+				annotation := namespace.Annotations[projectIdAnnotation]
+
+				if annotation == "" {
+					// no project is associated with namespace so don't do anything
+					logrus.Infof("no project is associated with namespace %v", namespaceName)
+					return nil
+				}
+
+				// TODO what if we can't parse this?
+				split := strings.Split(annotation, ":")
+				clusterName, projectId := split[0], split[1]
+
+				projects, err := m.projects.List("", labels.Everything())
+
+				if err != nil {
+					return fmt.Errorf("error getting projects: %v", err)
+				}
+
+				var project *v3.Project
+
+				for _, candidate := range projects {
+					if candidate.Name == projectId {
+						project = candidate
+					}
+				}
+
+				if project == nil {
+					return fmt.Errorf("no project with project id %v was found", projectId)
+				}
+
+				podSecurityPolicyTemplateId := project.Annotations[podSecurityPolicyTemplateLabel]
```

Comment:
should be getting it from project.spec.PodSecurityPolicyTemplateName, not from an attribute